### PR TITLE
[Remove Stream/Optional usage] LPS-175651 Remove_Stream_usages_for_module_sharepoint_soap_repository

### PR DESCRIPTION
**Pull request by jeffwu0724**
Hi Tina, Dante,

Resend: https://github.com/liferay-core-infra/liferay-portal/pull/3499
As talked, change to use toNode() for the case that only have on node. I decided to also keep toNodes() for the case that have more than one nodes. Because we still need the logic in _toNode(), and keeping toNodes() will be a better option than inlining the logic of _toNode() in the usage.
This is for ticket https://issues.liferay.com/browse/LPS-175651.
Thanks for checking!

### CODE REVIEW

**Comment by jeffwu0724** [comment]
ci:test:sharepoint

**Comment by zhangyan0701** [comment]
No regressions found.

File: modules/dxp/apps/sharepoint-soap/sharepoint-soap-repository/src/main/java/com/liferay/sharepoint/soap/repository/connector/internal/util/test/SharepointConnectionTestUtil.java [hunk 1]
```diff
@@ -80,9 +82,13 @@ private static String _getSharepointVMHostname() throws Exception {
 		try (BufferedReader bufferedReader = new BufferedReader(
 				new InputStreamReader(httpURLConnection.getInputStream()))) {
 
-			Stream<String> payloadStream = bufferedReader.lines();
+			List<String> data = new ArrayList<>();
 
-			String payload = payloadStream.collect(Collectors.joining());
+			while (bufferedReader.ready()) {
```
Comment (dantewang) [inline modules/dxp/apps/sharepoint-soap/sharepoint-soap-repository/src/main/java/com/liferay/sharepoint/soap/repository/connector/internal/util/test/SharepointConnectionTestUtil.java:87]: @jeffwu0724 Please use `while ((line = bufferedReader.readLine()) != null)` here. `ready()` only returns true when there is data in the buffer to be read. When reading from a network socket, it's common that the socket/inputstream is not closed, but there's no data to read -- the other end doesn't send any. In this case, `ready()` will return `false`.

Comment (dantewang) [reply modules/dxp/apps/sharepoint-soap/sharepoint-soap-repository/src/main/java/com/liferay/sharepoint/soap/repository/connector/internal/util/test/SharepointConnectionTestUtil.java:87]: @jeffwu0724 Also, please directly use a StringBundler.

**Pull request closed without merging**